### DISCUSSION TITLE
Access constituents corresponding to pT ordered jets?

### DISCUSSION THREAD

**Discussion by qibin2020** [scikit-hep/fastjet#320] [Q&A] [answered]
Dear experts,
I hope to ask how to read the `ClusterSequence.constituents()` of pT ordered jet?
I found I can use `sorted_by_pt()` to process the `inclusive_jets()`, while is there a way to handle the `constituents()` as well? Thank you!
(I am using the array interface)
Best,
Q.

**Comment by rkansal47** [comment]
Hi, is your question how to handle sort the constituents of a ClusterSequence by pT? 

The `ClusterSequence.constituents()` should return the particle 4-momenta, similar to [this list](https://github.com/scikit-hep/fastjet/blob/1a4f68c07197e7bcfeb43a057744dc303715f967/tests/test_002-exclusive_jets.py#L115-L121), but as an awkward array. Unfortunately, I see that you cannot directly use the `fastjet.sorted_by_pt()` function on this, but you can recreate the output array with the necessary behaviour and then use that function:

```python
import vector
arr = ak.Array(cluster.constituents(), behavior=vector.backends.awkward.behavior, with_name="Momentum4D")
sorted = fastjet.sorted_by_pt(arr)
```

Let us know if that doesn't answer your question. I will also let others comment if there are easier / more efficient solutions.

**Reply by qibin2020** [reply]
Hi thank you and basically I am wondering how to sort the constituents according to the jet pT. e.g. 
I can use fastjet.sorted_by_pt(inclusive_jets()) to get a list of sorted jet. Then I hope to access the constituents, in the order of this sorted jets.

**Reply by lgray** [reply, marked as answer]
```python3
constituents = ak.Array(cluster.constituents(), behavior=vector.backends.awkward.behavior, with_name="Momentum4D")
pt_idx = ak.argsort(constituents.pt, ascending=False) # highest pT first for each jet
pt_constituents = constituents[pt_idx]
```

is another solution.

**Reply by qibin2020** [reply]
Thank you all and yes that answer my question. I realized I can first get the order using `argsort()` and then sort in any axis. Also the directly `sorted_by_pt()` is compact and helpful. Thank you for your answers!
